src/rgw/driver/rados/rgw_bucket_sync.cc

```diff
@@ -590,8 +652,13 @@ void RGWBucketSyncFlowManager::reflect(const DoutPrefixProvider *dpp,
       pipe.source.apply_bucket(effective_bucket);
       pipe.dest.apply_bucket(effective_bucket);
 
-      ldpp_dout(dpp, 20) << __func__ << "(): flow manager (bucket=" << effective_bucket_key << "): adding source pipe: " << pipe << dendl;
-      source_pipes->insert(pipe);
+      if (is_forbidden) {
+        ldpp_dout(dpp, 20) << __func__ << "(): flow manager (bucket=" << effective_bucket_key << "): removing source pipe: " << pipe << dendl;
+        source_pipes->disable(pipe);
```

Comment:
@soumyakoduri do we have ordering issue, where we can enable stuff after we disabled that depending on the `flow_group` order?

Reply:
yes..while testing I noticed it. 

If there are two policies set, for eg.,
1) policy 1 - FORBIDDEN - <src_pipe,dest_pipe>
2) policy 2 - ENABLED - <src_pipe, dest_pipe>
policy2 is processed last in the order and thus granted the access.

To avoid that, have included `disabled_pipes` to store the restricted <pipe_set>s if any and match with the ones being inserted.